### PR TITLE
Port bugfix for chaos lost documents to 3.5.

### PR DESCRIPTION
**Pull request by neunhoef**
backport of https://github.com/arangodb/arangodb/pull/12545

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11815/

**Review by goedderz — CHANGES_REQUESTED** [review]
I'm missing the change in `arangod/Aql/EngineInfoContainerDBServerServerBased.cpp` that adds the rebootId to the request.

File: arangod/Cluster/Maintenance.cpp [hunk 1]
```diff
@@ -441,10 +474,10 @@ arangodb::Result arangodb::maintenance::diffPlanLocal(
     auto const& dbname = pdb.key.copyString();
     if (!local.hasKey(dbname)) {
       if (errors.databases.find(dbname) == errors.databases.end()) {
-        actions.emplace_back(
-            ActionDescription({{std::string(NAME), std::string(CREATE_DATABASE)},
-                               {std::string(DATABASE), std::string(dbname)}},
-                              HIGHER_PRIORITY));
+        actions.emplace_back(ActionDescription(
+              std::map<std::string, std::string>{{std::string(NAME), std::string(CREATE_DATABASE)}, {std::string("tick"), std::to_string(TRI_NewTickServer())},
```
Comment (jsteemann) [inline arangod/Cluster/Maintenance.cpp:478]:
```suggestion
              std::map<std::string, std::string>{{std::string(NAME), std::string(CREATE_DATABASE)},
```

File: arangod/Cluster/Maintenance.cpp [hunk 2]
```diff
@@ -456,10 +489,10 @@ arangodb::Result arangodb::maintenance::diffPlanLocal(
   for (auto const& ldb : VPackObjectIterator(local)) {
     auto const& dbname = ldb.key.copyString();
     if (!plan.hasKey(std::vector<std::string>{DATABASES, dbname})) {
-      actions.emplace_back(
-          ActionDescription({{std::string(NAME), std::string(DROP_DATABASE)},
-                             {std::string(DATABASE), std::string(dbname)}},
-                            HIGHER_PRIORITY));
+      actions.emplace_back(ActionDescription(
+          std::map<std::string, std::string>{{std::string(NAME), std::string(DROP_DATABASE)}, {std::string("tick"), std::to_string(TRI_NewTickServer())},
```
Comment (jsteemann) [inline arangod/Cluster/Maintenance.cpp:493]:
```suggestion
          std::map<std::string, std::string>{{std::string(NAME), std::string(DROP_DATABASE)}, 
```

**Review by jsteemann — APPROVED** [review]

**Comment by jsteemann** [comment]
@goedderz : I added the missing part. Can you recheck? Thanks!

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11872/

**Review by goedderz — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11875/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11881/

**Comment by jsteemann** [comment]
Tests blue

**Review by KVS85 — APPROVED** [review]

**Pull request merged**